packages/assets-controllers/src/selectors.ts

```diff
@@ -0,0 +1,100 @@
+
+const tokensInformations = (state: TokensControllerState) => state.allTokens;
+
+export const evmAggregatedBalancesForAccount = createSelector(
+  evmTokenBalances,
+  evmTokenRates,
+  currencyConversionRate,
+  tokensInformations,
+  (_state: TokenBalancesControllerState, account: Hex) => account,
+  (
+    tokenBalances,
+    tokenRates,
+    currencyConversionRateState,
+    tokensInformationsState,
+    account,
+  ) => {
+    const balancesForAccount = tokenBalances?.[account];
+    if (!balancesForAccount) {
+      return {};
+    }
+
+    // Calculate total balance value across all chains for this account
+    let totalBalance = 0;
+
+    for (const [chainId, chainBalances] of Object.entries(balancesForAccount)) {
+      for (const [tokenAddress, balance] of Object.entries(chainBalances)) {
+        // tokenRates is organized by chainId -> tokenAddress -> MarketDataDetails
+        // We need to find the market data for this token across all chains
+        let marketData;
+        for (const chainData of Object.values(tokenRates)) {
+          if (chainData[tokenAddress as Hex]) {
+            marketData = chainData[tokenAddress as Hex];
+            break;
+          }
+        }
+
+        const rateToUse =
+          currencyConversionRateState[marketData?.currency ?? '']
+            ?.conversionRate ?? 0;
+
+        if (marketData?.price) {
+          const { price } = marketData;
+
+          // Find the token decimals from tokensInformations
+          let tokenDecimals = 18; // Default to 18 decimals
+          const chainTokens = tokensInformationsState[chainId as Hex];
+          if (chainTokens) {
+            // Look through all accounts in this chain to find the token
+            for (const accountTokens of Object.values(chainTokens)) {
+              if (Array.isArray(accountTokens)) {
+                const foundToken = accountTokens.find(
+                  (token: { address: string; decimals: number }) =>
+                    token.address.toLowerCase() === tokenAddress.toLowerCase(),
+                );
+                if (foundToken && foundToken.decimals !== undefined) {
+                  tokenDecimals = foundToken.decimals;
+                  break;
+                }
+              }
+            }
+          }
+
+          // Convert all balances to 18 decimal equivalent first
+          let normalizedBalance = convertHexToDecimal(balance);
+
+          // If token has different decimals, convert to 18 decimals
+          if (tokenDecimals !== 18) {
+            const decimalDifference = 18 - tokenDecimals;
+            normalizedBalance *= Math.pow(10, decimalDifference);
+          }
+
+          // Now divide by 10^18 since everything is normalized to 18 decimals
+          const balanceInTokens = normalizedBalance / Math.pow(10, 18);
+          const tokenValue = price * balanceInTokens * rateToUse;
+          const roundedTokenValue = Math.round(tokenValue * 100) / 100; // Round to 2 decimal places
```

Comment:
I might need you to break down the formula again.

- `balanceInTokens` - e.g. 1MNT
- `rateToUse` - ETH conversion rate - e.g. 0.0032...ETH = 1MNT.
- `price` - USD/Currency Price - e.g. $1234 = 1ETH